modules/netweaver/main.tf

```diff
@@ -0,0 +1,134 @@
+}
+
+resource "google_compute_disk" "gcp_nw_pd_1" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-1"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.sap_mnt_size}"
+}
+
+resource "google_compute_disk" "gcp_nw_pd_2" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-2"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.swap_size}"
+}
+
+resource "google_compute_attached_disk" "gcp_nw_attached_pd_0" {
+  project     = "${var.project_id}"
+  device_name = "${var.usr_sap_size > 0 ? local.device_name_1 : ""}"
+  disk        = "${google_compute_disk.gcp_nw_pd_0.self_link}"
+  instance    = "${google_compute_instance.gcp_nw.self_link}"
+}
+
+resource "google_compute_attached_disk" "gcp_nw_attached_pd_1" {
+  project     = "${var.project_id}"
+  device_name = "${var.sap_mnt_size > 0 ? local.device_name_2 : ""}"
+  disk        = "${google_compute_disk.gcp_nw_pd_1.self_link}"
+  instance    = "${google_compute_instance.gcp_nw.self_link}"
+}
+
+resource "google_compute_attached_disk" "gcp_nw_attached_pd_2" {
+  project     = "${var.project_id}"
+  device_name = "${var.swap_size > 0 ? local.device_name_3 : ""}"
+  disk        = "${google_compute_disk.gcp_nw_pd_2.self_link}"
+  instance    = "${google_compute_instance.gcp_nw.self_link}"
+}
+
+resource "google_compute_instance" "gcp_nw" {
+  project                   = "${var.project_id}"
+  name                      = "${var.instance_name}"
+  machine_type              = "${var.instance_type}"
+  zone                      = "${var.zone}"
+  tags                      = "${var.network_tags}"
+  allow_stopping_for_update = true
+  can_ip_forward            = true
+
+  scheduling {
+    automatic_restart   = true
+    on_host_maintenance = "MIGRATE"
+  }
+
+  boot_disk {
+    auto_delete = "${var.autodelete_disk}"
+
+    device_name = "${var.instance_name}-${var.device_0}"
+
+    initialize_params {
+      image = "projects/${var.linux_image_project}/global/images/family/${var.linux_image_family}"
+      size  = "${var.boot_disk_size}"
+      type  = "${var.boot_disk_type}"
+    }
+  }
+
+  network_interface {
+    subnetwork         = "${var.subnetwork}"
+    subnetwork_project = "${var.project_id}"
+    access_config      = "${local.access_config[var.public_ip]}"
+  }
+
+  metadata {
+    instanceName           = "${var.instance_name}"
+    instanceType           = "${var.instance_type}"
+    post_deployment_script = "${var.post_deployment_script}"
+    subnetwork             = "${var.subnetwork}"
+    usrsapSize             = "${var.usr_sap_size}"
+    sapmntSize             = "${var.sap_mnt_size}"
+    swapSize               = "${var.swap_size}"
+    sap_deployment_debug   = "${var.sap_deployment_debug}"
+    startup-script         = "${var.startup_script}"
+    publicIP               = "${var.public_ip}"
+  }
+
+  service_account {
+    email = "${var.service_account_email}"
+    scopes = [
```

Comment:
As I mentioned earlier, access scopes are considered to be legacy. This list should be empty and the required IAM roles for the service account should be documented in the README.

Reply:
These are the set up scopes as stated in the Deployment Manager scripts since we are trying to mimic whats there in DM to Terraform.
https://storage.googleapis.com/sapdeploy/dm-templates/sap_nw/sap_nw.py
![image](https://user-images.githubusercontent.com/17788117/64052188-76983280-cb32-11e9-909b-2aa174868088.png)

Reply:
even the customers and testers are demanding to show it like this, since they are not interested in creating IAM Roles while testing or developing these modules.

Reply:
Who are the 'testers' you're referring to?

Google PSO needs to follow and promote best practices, which includes privilege service accounts. We can and should make generating the correct Service Accounts easy, but should use the right IAM roles.

Reply:
Hi Morgante,
I totally agree that we need IAM roles instead of legacy roles. Have asked Ajay and his testing team about it . Awaiting their response.

Reply:
made those changes and used IAM roles instead of legacy rules

Reply:
made changes to the code and added the scope ="cloud-platform"